test/validation/api/crypto/odp_crypto_test_inp.c

```diff
@@ -1219,12 +1220,14 @@ static void create_hash_test_reference(odp_auth_alg_t auth,
 
 	session = session_create(ODP_CRYPTO_OP_ENCODE, ODP_CIPHER_ALG_NULL,
 				 auth, ref, PACKET_IV, HASH_NO_OVERLAP);
+	if (session == ODP_CRYPTO_SESSION_INVALID)
+		return -1;
 
 	if (crypto_op(pkt, &ok, session,
 		      ref->cipher_iv, ref->auth_iv,
 		      &cipher_range, &auth_range,
 		      ref->aad, enc_digest_offset))
-		return;
+		return 0;
```

Comment:
This change is basically correct for the change being done, but I think it would be better to CU_ASSERT that the return value is zero and then continue the function (then the session would be freed too). I think the code should have been made that way in the first place but now that you are touching it anyway, you could fix it.

Reply:
Ack. Created separate commit for the second fix and the suggested change is incorporated in it.